### PR TITLE
Community refresh 1.1.0 

### PR DESCRIPTION
**Pull request by iArpanPatel**
Community refresh for 1.1.0 changes.
1. Script changes:
2. Documentation changes:
    Content related to Swarm Learning 1.1.0 is updated:
    - How to Swarm enable an ML algorithm section updated for default values for useAdaptiveSync and mlPlatform.
    - Swarm Learning component interactions section updated.
    - Swarm Learning Command Interface section updated for WAIT FOR TASKRUNNER and SLEEP.
    - SWCI APIs section updated for sleep () and an example snippet is added at the end of the section for Swci().
    - Examples section updated for all four examples.
    - Frequently asked questions section updated with two new FAQ's.
    - Environment variables section updated for SL_WAIT_FOR_FULL_QUORUM_SECONDS, SWCI_TASK_MAX_WAIT_TIME, and
       SWOP_KEEP_CONTAINERS
    - Install HPE Swarm Learning section updated.
    - Running Swarm Learning section updated for primary and secondary APLS IP and Port information.
    - Starting SWCI nodes section updated with a note on swci-init script file.
    - Upgrading Swarm Learning section added.

3. Examples changes:
    - Update for creating Volume and copy whl file using hello-world, remove volume before creating.
    - Sleep command in swci-init in applicable examples. 
    - Cifar10 cleanup updates

### CODE REVIEW

File: docs/User/Swarm_Learning_Command_Interface.md [hunk 1]
```diff
@@ -68,7 +68,7 @@ A SWCI tool can control operations on multiple Swarm Learning framework instance
...
 |`LIST TASKRUNNER PEERS <taskRunnerName:string>`|Displays list of enrolled peers for the specified Taskrunner ID.|
-|`WAIT FOR TASKRUNNER <trName : string>`|Waits for the Taskrunner to complete its current task.<br>`trName`: Specifies the unique name of the Taskrunner.<br>Prints a message periodically until the task completes. When the task completes, the final state of the task is printed.<br>|
+|`WAIT FOR TASKRUNNER <trName : string>`|Waits for the Taskrunner to complete its current task.<br>`trName`: Specifies the unique name of the Taskrunner.<br>WAITING FOR TASKRUNNER TO COMPLETE - Maximum wait time is : <SWCI_TASK_MAX_WAIT_TIME>.<br>If SWCI_TASK_MAX_WAIT_TIME is not set, the default value is 120 mins.<br>Prints # (a progress indicator) periodically until the task completes.<br>When the task completes, the final state of the task is printed.<br>If the maximum wait time is reached, the following warning message is displayed:<br>WARNING - Maximum configured SWCI wait time is over|
```
Comment (iArpanPatel) [inline docs/User/Swarm_Learning_Command_Interface.md:71]: Make Code block wherever applicable.
Line spacing appropriately.

File: docs/User/Swarm_Learning_Command_Interface.md [hunk 2]
```diff
@@ -77,7 +77,7 @@ A SWCI tool can control operations on multiple Swarm Learning framework instance
...
 |`EXIT`|This command exits the SWCI session unconditionally.|
 |`EXIT ON FAILURE \[ON/OFF\]`|This command instructs SWCI to exit the current session when any of the subsequent commands fail. The default value is OFF.|
-|`LIST NODES`|This command prints the list of registered and active nodes.|
+|`SLEEP`|This command sleeps for a specified time before executing the subsequent commands.<br>For example, in between a WAIT FOR TASKRUNNER and RESET TASKRUNNER, one can use a SLEEP 10, to give a grace time of 10 secs, before the RESET command cleans up the SL and user container.<br>This would be required to allow the user ML code to save the model or do any inference of the model, after the Swarm training is over.<br>For more information, see the example SWCI scripts in the swarm-learning/examples/ directory.|
```
Comment (iArpanPatel) [inline docs/User/Swarm_Learning_Command_Interface.md:80]: Code blocks, Line spacing

File: docs/User/SWCI_APIs.md [hunk 3]
```diff
@@ -43,6 +43,40 @@ The python3 program needs to import the SWCI class, and then use the below APIs.
...
 |`resetTaskRunner()`|This method resets the state of the taskrunner contract to an uninitialized state.<br><strong>WARNING:</strong>This action cannot be undone, reset only completed Taskrunner contracts. Resetting the active taskrunner contract can result in unexpected behavior.|`trName='defaulttaskbb.taskdb.sml.hpe'`|
 |`resetTrainingContract()`|This method resets the state of the training contract to an uninitialized state.<br><strong>WARNING:</strong>This action cannot be undone, reset only completed Swarm Learning contracts. Resetting the active contracts can result in unexpected behavior.|`ctName='defaultbb.cqdb.sml.hpe'`|
+|`sleep()`|This method sleeps for a specified time before executing the subsequent commands.<br>For example, in between a WAIT FOR TASKRUNNER and RESET TASKRUNNER, one can use a SLEEP 10, to give a grace time of 10 secs, before the RESET command cleans up the SL and user container.<br>This would be required to allow the user ML code to save the model or do any inference of the model, after the Swarm training is over. <br>For more information, see the example SWCI scripts in the  swarmlearning/examples/ directory.|`time in seconds`|
```
Comment (iArpanPatel) [inline docs/User/SWCI_APIs.md:46]: code blocks, line spacing

File: docs/User/SWCI_APIs.md [hunk 4]
```diff
@@ -43,6 +43,40 @@ The python3 program needs to import the SWCI class, and then use the below APIs.
...
+# 1. Swarm Learning Infrastructure is setup and ready.
+# 2. SWCI container is running in WEB mode (-e SWCI_MODE='WEB')
+# 3. There should be explicit port forwarding for SWCI_WEB_PORT while running the SWCI container (ex: -p 30306:30306)
+# 4. Swarm learning wheel package should be installed in the environment where we run this file.
+##################################################################################################################
+```
+
+```
+# Import swci from the swarmlearning whl package
+import swarmlearning.swci as sw
```
Comment (iArpanPatel) [inline docs/User/SWCI_APIs.md:67]: Line spacing

File: examples/fraud-detection/Credit_card_fraud_detection.md [hunk 5]
```diff
@@ -72,9 +72,10 @@ mv workspace/fraud-detection/data-and-scratch workspace/fraud-detection/user4/
 6.  Create a docker volume and copy Swarm Learning wheel file.
 
 ```
+docker volume rm sl-cli-lib
 docker volume create sl-cli-lib
-docker container create --name helper -v sl-cli-lib:/data hub.myenterpriselicense.hpe.com/hpe_eval/swarm-learning/sn:1.0.0
-docker cp -L lib/swarmlearning-client-py3-none-manylinux_2_24_x86_64.whl helper:/data
+docker container create --name helper -v sl-cli-lib:/data hello-world
+docker cp lib/swarmlearning-client-py3-none-manylinux_2_24_x86_64.whl helper:/data
 docker rm helper
```
Comment (iArpanPatel) [inline examples/fraud-detection/Credit_card_fraud_detection.md:80]: Add note for the Symbolic link step

File: examples/cifar10/CIFAR-10.md [hunk 6]
```diff
@@ -104,7 +104,7 @@ docker build -t user-ml-env-tf2.7.0 --build-arg https_proxy=http://<your-proxy-s
 --ml-image=user-ml-env-tf2.7.0 --ml-name=ml1 \
 --ml-w=/tmp/test --ml-entrypoint=python3 --ml-cmd=model/cifar10.py \
 --ml-v=workspace/cifar10/model:/tmp/test/model \
---ml-e DATA_DIR=app-data --ml-e MODEL_DIR=model \
+--ml-e MODEL_DIR=model \
```
Comment (iArpanPatel) [inline examples/cifar10/CIFAR-10.md:107]: In step 5, create symbolic link for whl file. 
Also change the whl file version to 1.1.0

File: examples/fraud-detection/Credit_card_fraud_detection.md [hunk 7]
```diff
@@ -72,9 +72,10 @@ mv workspace/fraud-detection/data-and-scratch workspace/fraud-detection/user4/
 6.  Create a docker volume and copy Swarm Learning wheel file.
 
 ```
+docker volume rm sl-cli-lib
 docker volume create sl-cli-lib
-docker container create --name helper -v sl-cli-lib:/data hub.myenterpriselicense.hpe.com/hpe_eval/swarm-learning/sn:1.0.0
-docker cp -L lib/swarmlearning-client-py3-none-manylinux_2_24_x86_64.whl helper:/data
+docker container create --name helper -v sl-cli-lib:/data hello-world
+docker cp lib/swarmlearning-client-py3-none-manylinux_2_24_x86_64.whl helper:/data
```
Comment (iArpanPatel) [inline examples/fraud-detection/Credit_card_fraud_detection.md:78]: docker cp -L option correction.

File: examples/mnist-pyt/MNIST-PYT.md [hunk 8]
```diff
@@ -107,9 +107,10 @@ sed -i "s+<CURRENT-PATH>+$(pwd)+g" workspace/mnist-pyt/swop/swop*_profile.yaml w
 7.  On both host-1 and host-2, create a docker volume and copy Swarm Learning wheel file:
 
 ```
+docker volume rm sl-cli-lib
 docker volume create sl-cli-lib
-docker container create --name helper -v sl-cli-lib:/data hub.myenterpriselicense.hpe.com/hpe_eval/swarm-learning/sn:1.0.0
-docker cp -L lib/swarmlearning-client-py3-none-manylinux_2_24_x86_64.whl helper:/data
+docker container create --name helper -v sl-cli-lib:/data hello-world
+docker cp lib/swarmlearning-client-py3-none-manylinux_2_24_x86_64.whl helper:/data
```
Comment (iArpanPatel) [inline examples/mnist-pyt/MNIST-PYT.md:113]: docker cp -L option correction.
Note for Pre requisite for symbolic link.

File: examples/mnist/MNIST.md [hunk 9]
```diff
@@ -86,9 +86,10 @@ sed -i "s+<CURRENT-PATH>+$(pwd)+g" workspace/mnist/swop/swop*_profile.yaml works
...
 
 ```
+docker volume rm sl-cli-lib
 docker volume create sl-cli-lib
-docker container create --name helper -v sl-cli-lib:/data hub.myenterpriselicense.hpe.com/hpe_eval/swarm-learning/sn:1.0.0
-docker cp -L lib/swarmlearning-client-py3-none-manylinux_2_24_x86_64.whl helper:/data
+docker container create --name helper -v sl-cli-lib:/data hello-world
+docker cp lib/swarmlearning-client-py3-none-manylinux_2_24_x86_64.whl helper:/data
 docker rm helper
 ```
```
Comment (iArpanPatel) [inline examples/mnist/MNIST.md:94]: Note for pre requisite symbolic link. 
docker cp -L option correction.

File: docs/User/Frequently_asked_questions.md [hunk 10]
```diff
@@ -170,3 +170,14 @@ Yes. New nodes can be added in the network at any point in the training. Just li
 
 Swarm Learning uses averaging as the merge algorithm. Currently, users cannot specify the merge algorithm. This will be supported in a later release.
 
+## Before enabling Swarm Learning, how to confirm the standalone user application has no issues and runs?
+
+Run the user container with SWARM_LOOPBACK set to TRUE, this bypasses Swarm Learning to help you quickly develop, integrate, and test your model code with Swarm Learning package. If your code runs to completion and saves the local model it would indicate that the ML application may not have any issues.
```
Comment (iArpanPatel) [inline docs/User/Frequently_asked_questions.md:175]: code blocks, line spacing

File: docs/Install/Environment_variables.md [hunk 11]
```diff
@@ -10,11 +10,13 @@ The environment variables are passed to containers or added to the environment v
 |`SN_ETH_PORT_EXT`|Sets an Ethernet port for Swarm Network node.|
 |`SN_I_AM_SENTINEL`| Sets a Swarm Network node to become the Sentinel node, only when it is set to true.<br> Default value: False<br> |
 |`SN_START_MINING`| Starts mining on non-sentinel nodes. \(Optional\)<br> Default value: False<br> |
-|`SL_WAIT_FOR_FULL_QUORUM_SECONDS`|Sets maximum time to wait for full quorum before an SL node, designated as leader node, decides to use minPeers nodes.|
+|`SL_WAIT_FOR_FULL_QUORUM_SECONDS`|Sets the maximum time for an SL leader node to wait for full quorum after minPeers are ready for merge. This parameter lets you to maximize the number of peers participating in the merge process.|
```
Comment (iArpanPatel) [inline docs/Install/Environment_variables.md:13]: Default Value missed out.

File: docs/Install/Environment_variables.md [hunk 12]
```diff
@@ -10,11 +10,13 @@ The environment variables are passed to containers or added to the environment v
...
 |`SN_START_MINING`| Starts mining on non-sentinel nodes. \(Optional\)<br> Default value: False<br> |
-|`SL_WAIT_FOR_FULL_QUORUM_SECONDS`|Sets maximum time to wait for full quorum before an SL node, designated as leader node, decides to use minPeers nodes.|
+|`SL_WAIT_FOR_FULL_QUORUM_SECONDS`|Sets the maximum time for an SL leader node to wait for full quorum after minPeers are ready for merge. This parameter lets you to maximize the number of peers participating in the merge process.<br>Default value: 30 secs|
+|`SWCI_TASK_MAX_WAIT_TIME`|Specifies a maximum timeout value for the completion of a task.<br>This value must be set in minutes, and the default is 120 mins (2 hours).
 |`SWCI_MODE`| Enables SWCIs web interface instead of command line interface. Allowed values are CLI and WEB.<br> Default value: CLI<br> |
 |`SWCI_STARTUP_SCRIPT`|This is a default start script of SWCI.|
 |`SWCI_WEB_PORT`|Default port on which SWCI-WEB starts server.|
 |`SWOP_PROFILE`|Indicates default profile for SWOP.|
+|`SWOP_KEEP_CONTAINERS`|SL and ML containers spawned by SWOP are not removed. This option can be enabled to retain the stopped containers for debugging.<br>Default value: False|
```
Comment (girisairam450) [inline docs/Install/Environment_variables.md:19]: Maybe rephrase it for better understanding. Something like By default, SL and ML containers will be removed. Instead of mentioning not removed

File: docs/Install/Install_HPE_Swarm_Learning.md [hunk 13]
```diff
@@ -1,44 +1,35 @@
 # <a name="GUID-60017971-B0A9-4119-AEAF-A21594EE5C1E"/> Install HPE Swarm Learning
 
-1.   After [completing installation of license server and downloading of the Swarm Learning installation files](/docs/Install/Install_the_License_Server.md), you must run one of the Swarm Learning installer based on your platform (Linux, Windows, or Mac as listed below). 
+1.   After [completing installation of license server and downloading of the Swarm Learning installation files](/docs/Install/Install_the_License_Server.md), you must
+run one of the Swarm Learning installer based on your platform (Linux, Windows, or Mac as listed below). 
 
      - HPE_SWARM_LEARNING_INSTALLER_LINUX_Q2V41-11036
      - HPE_SWARM_LEARNING_INSTALLER_WINDOWS_Q2V41-11038.EXE
```
Comment (girisairam450) [inline docs/Install/Install_HPE_Swarm_Learning.md:7]: check if it is .EXE or .exe

**Review by girisairam450 — APPROVED** [review]

**Review by htjain — APPROVED** [review]

File: docs/Install/Install_HPE_Swarm_Learning.md (hunk 13 above)
Comment (iArpanPatel) [reply docs/Install/Install_HPE_Swarm_Learning.md]: corrected

File: docs/User/Swarm_Learning_Command_Interface.md (hunk 1 above)
Comment (iArpanPatel) [reply docs/User/Swarm_Learning_Command_Interface.md]: Corrected.

File: docs/User/Swarm_Learning_Command_Interface.md (hunk 2 above)
Comment (iArpanPatel) [reply docs/User/Swarm_Learning_Command_Interface.md]: Corrected

File: docs/Install/Install_HPE_Swarm_Learning.md [hunk 14]
```diff
@@ -1,44 +1,35 @@
...
 
-     -logs
-     :   If enabled, displays the detail message on the CLI during the installation. To enable, use the command, `-logs verbose`.
+     **-logs**
+     :    If enabled, displays the detail message on the CLI during the installation.<br> To enable, use the command, `-logs verbose`.
 
-     -version
-     :   This option defines the version of docker images that must be installed. The default value is 1.0.0. Example, `-version 0.3.0`
+     **-version**
+     :    This option defines the version of docker images that must be installed. The default value is 1.0.0.<br> Example, `-version 0.3.0`
```
Comment (suresh-ls) [inline docs/Install/Install_HPE_Swarm_Learning.md:26]: default value is 1.1.0

File: docs/User/SWCI_APIs.md (hunk 3 above)
Comment (iArpanPatel) [reply docs/User/SWCI_APIs.md:46]: corrected

File: docs/User/SWCI_APIs.md (hunk 4 above)
Comment (iArpanPatel) [reply docs/User/SWCI_APIs.md:67]: corrected

File: examples/fraud-detection/Credit_card_fraud_detection.md (hunk 5 above)
Comment (iArpanPatel) [reply examples/fraud-detection/Credit_card_fraud_detection.md:80]: corrected

File: examples/fraud-detection/Credit_card_fraud_detection.md (hunk 7 above)
Comment (iArpanPatel) [reply examples/fraud-detection/Credit_card_fraud_detection.md]: corrected

File: examples/mnist-pyt/MNIST-PYT.md (hunk 8 above)
Comment (iArpanPatel) [reply examples/mnist-pyt/MNIST-PYT.md]: corrected

File: examples/mnist/MNIST.md (hunk 9 above)
Comment (iArpanPatel) [reply examples/mnist/MNIST.md:94]: corrected

File: docs/Install/Environment_variables.md (hunk 11 above)
Comment (iArpanPatel) [reply docs/Install/Environment_variables.md]: added default value

File: docs/Install/Environment_variables.md (hunk 12 above)
Comment (iArpanPatel) [reply docs/Install/Environment_variables.md]: Rephrased "By default, SL and ML containers spawned by SWOP are removed"

File: docs/Install/Install_HPE_Swarm_Learning.md (hunk 13 above)
Comment (iArpanPatel) [reply docs/Install/Install_HPE_Swarm_Learning.md]: corrected

File: docs/Install/Starting_SWCI_nodes.md [hunk 15]
```diff
@@ -23,4 +23,12 @@ The run-swci script accepts the following parameters:
 |`--usr-dir <dir>`|The host directory that must be used as the user directory by this SWCI node.|None|
 |`--init-script-name <swci-init file>`*|Name of the init script file that has SWCI commands to be executed at the start of SWCI. <br> This file must be located inside the user directory at the top level.<br>|`swci-init`|
 
-*If init option is provided, all SWCI commands within this script file are processed before it enters the interactive mode and waits for users commands. Users can simulate a non-interactive SWCI run by having a bunch of SWCI commands and an SWCI `EXIT` command at the end of the `swci-init` file. This could be used for automation.
+<blockquote>
```
Comment (suresh-ls) [inline docs/Install/Starting_SWCI_nodes.md:26]: Good to retain the old text like this, before the Note: section...  It adds some value: 

"If init script option is provided, all SWCI commands within this script file are processed before SWCI enters the interactive mode and waits for users commands. Users can simulate a non-interactive SWCI run by having a bunch of SWCI commands and an SWCI `EXIT` command at the end of the `swci-init` file. This could be used for automation."

File: docs/Install/Install_HPE_Swarm_Learning.md (hunk 14 above)
Comment (iArpanPatel) [reply docs/Install/Install_HPE_Swarm_Learning.md]: corrected

File: docs/Install/Versioning_and_upgrade.md [hunk 16]
```diff
@@ -1,4 +1,6 @@
 # <a name="GUID-2E350669-7E5A-47BC-AB15-58AC4CFAD9C1"/> Upgrading from earlier evaluation version
 
-The current Swarm Learning release is not compatible with the earlier Eval.0.3.0 release. **You must** uninstall the earlier version before installing the current HPE Swarm Learning version.
+To upgrade to a latest version, see [Install HPE Swarm Learning](Install_HPE_Swarm_Learning.md).
```
Comment (suresh-ls) [inline docs/Install/Versioning_and_upgrade.md:3]: Why is this left out ? - 
"The current HPE Swarm Learning release is not compatible with the Eval or community 0.3.0 release. You must uninstall the 0.3.0 release before installing the current HPE Swarm Learning version."

File: docs/User/Swarm_Learning_Command_Interface.md [hunk 17]
```diff
@@ -68,7 +68,7 @@ A SWCI tool can control operations on multiple Swarm Learning framework instance
...
 |`LIST TASKRUNNER PEERS <taskRunnerName:string>`|Displays list of enrolled peers for the specified Taskrunner ID.|
-|`WAIT FOR TASKRUNNER <trName : string>`|Waits for the Taskrunner to complete its current task.<br>`trName`: Specifies the unique name of the Taskrunner.<br>Prints a message periodically until the task completes. When the task completes, the final state of the task is printed.<br>|
+|`WAIT FOR TASKRUNNER <trName : string>`|Waits for the Taskrunner to complete its current task.<br>-`trName`: Specifies the unique name of the Taskrunner.<br>-`WAITING FOR TASKRUNNER TO COMPLETE` - Maximum wait time is : `<SWCI_TASK_MAX_WAIT_TIME>`.<br>-If `SWCI_TASK_MAX_WAIT_TIME` is not set, the default value is 120 mins.<br>Prints # (a progress indicator) periodically until the task completes. When the task completes, the final state of the task is printed.<br>If the maximum wait time is reached, the following warning message is displayed:<br>WARNING - Maximum configured SWCI wait time is over|
```
Comment (suresh-ls) [inline docs/User/Swarm_Learning_Command_Interface.md:71]: We should have hypen only for the first one..  This command has only one option..

Let's remove the 2 other hypens...

Comment (iArpanPatel) [reply docs/User/Swarm_Learning_Command_Interface.md]: Removed extra hyphens and added line spacing for readability.

File: docs/Install/Versioning_and_upgrade.md (hunk 16 above)
Comment (iArpanPatel) [reply docs/Install/Versioning_and_upgrade.md:4]: Restored and minor formatting done.

File: docs/User/Frequently_asked_questions.md (hunk 10 above)
Comment (iArpanPatel) [reply docs/User/Frequently_asked_questions.md:175]: Corrected

**Review by suresh-ls — APPROVED** [review]

File: docs/User/How_to_Swarm_enable_an_ML_algorithm.md [hunk 18]
```diff
@@ -38,13 +38,13 @@ swarmCallback.logger.setLevel(logging.DEBUG)
 |---------|-----------|
 |`syncFrequency`|Specifies the number of batches of local training to be performed between two swarm sync rounds. If adaptive sync enabled, this is the frequency to be used at the start.|
 |`minPeers`|Specifies the minimum number of SL peers required during each synchronization round for Swarm Learning to proceed further.|
-|`useAdaptiveSync`|Modulate the next syncFrequency value post each synchronization round based on performance on validation data.|
+|`useAdaptiveSync`|Modulate the next syncFrequency value post each synchronization round based on performance on validation data. The default value is false.|
```
Comment (suresh-ls) [inline docs/User/How_to_Swarm_enable_an_ML_algorithm.md:41]: Can we add this:   

Note: As of now, this option is available only for KERAS

File: docs/User/Swarm_Learning_component_interactions.md [hunk 19]
```diff
@@ -1,6 +1,6 @@
 # <a name="GUID-CE2496F4-22BD-468B-AD40-011E3F113E6E"/> Swarm Learning component interactions
 
-The Swarm nodes interact with each other in many ways using network ports that are dedicated for each purpose:![Swarm component interactions](GUID-7DB64042-89CD-49D1-9728-A93739BF8816-high.png)
+The Swarm nodes interact with each other in many ways using network ports that are dedicated for each purpose:![Swarm component interactions](Swarm_Learning_component_interaction.png)
```
Comment (suresh-ls) [inline docs/User/Swarm_Learning_component_interactions.md:3]: Interaction 6: In the table below, we need to add the FIFO details from PDF file..

File: docs/User/How_to_Swarm_enable_an_ML_algorithm.md (hunk 18 above)
Comment (iArpanPatel) [reply docs/User/How_to_Swarm_enable_an_ML_algorithm.md]: Added Note

File: docs/User/Swarm_Learning_component_interactions.md (hunk 19 above)
Comment (iArpanPatel) [reply docs/User/Swarm_Learning_component_interactions.md:3]: Added FIFO details

**Review by htjain — APPROVED** [review]

**Review by suresh-ls — APPROVED** [review]
